preview/Win7Msix/Win7MSIXInstaller/FileTypeAssociation.cpp

```diff
@@ -142,11 +142,21 @@ HRESULT FileTypeAssociation::ParseManifest()
 
 HRESULT FileTypeAssociation::ExecuteForAddRequest()
 {
+    if (m_msixRequest->GetMsixResponse()->GetIsInstallCancelled())
+    {
+        return HRESULT_FROM_WIN32(ERROR_INSTALL_USEREXIT);
+    }
+
     for (std::vector<Fta>::iterator fta = m_Ftas.begin(); fta != m_Ftas.end(); ++fta)
     {
         RETURN_IF_FAILED(ProcessFtaForAdd(*fta));
     }
 
+    if (m_msixRequest->GetMsixResponse()->GetIsInstallCancelled())
+    {
+        return HRESULT_FROM_WIN32(ERROR_INSTALL_USEREXIT);
+    }

```

Comment:
i don't think we need this to be checked before AND after each handler because there is hardly any work between handlers and we expect each handler to be checking it. However, it might make sense to do it after each FTA (i.e. inside the loop) in case each FTA has significant work.

Reply:
Updated check for FTA handler and added check inside loop. Also, updated the same for Protocol, StartMenulink and AddRemovePrograms.